platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -298,7 +303,105 @@ namespace platoon_strategic_ihp
+         * 
+         * @param
+         * start_downtrack: The current downtrack distance, in m.
+         * end_downtrack: The target downtrack distance in the target lane, in m.
+         * 
+         * @return
+         * target_lanelet_id: The lanelet id of the target lane.
+         */
+
+        std::vector<lanelet::ConstLanelet> lanelets_in_path = wm_->getLaneletsBetween(start_downtrack, end_downtrack, false, false);
+        std::vector<double> downtracks,crosstracks;
+        std::vector<lanelet::ConstLanelet> lanelets_samelane,lanelets_sidelane;
+        lanelet::ConstLanelet target_lanelet;
+        int target_lanelet_id;
+        if (!lanelets_in_path.empty())
+        {
+            for (int i = 0; i < lanelets_in_path.size(); ++i)
+            {
+                lanelet::BasicPoint2d p1=lanelets_in_path[i].centerline2d().basicLineString().back();
+                double downtrack=wm_->routeTrackPos(p1).downtrack;
+                double crosstrack=wm_->routeTrackPos(p1).crosstrack;
+                if (abs(crosstrack)>1)//1 is a threshold, the assumption for this condition is that there are two lanes on the road
+                {
+                    lanelets_sidelane.push_back(lanelets_in_path[i]);
+                    downtracks.push_back(downtrack);
+                    crosstracks.push_back(crosstrack);
+                }else
+                {
+                    lanelets_samelane.push_back(lanelets_in_path[i]);
+                }
+            }
+            if (!downtracks.empty())
+            {
+                int maxPosition = max_element(downtracks.begin(),downtracks.end()) - downtracks.begin();
+                target_lanelet=lanelets_sidelane[maxPosition];
+                target_lanelet_id=target_lanelet.id();
+            }
+        }
+        return target_lanelet_id;
+    } 
+
+    // Check if the platoon close to the platoon for cut-in join.
+    bool PlatoonStrategicIHPPlugin::isVehicleNextLane(double frontVehicleDtd, double rearVehicleDtd, double frontVehicleCtd)
+    {
+        /**
+         * @brief Function to determine if the host vehicle is close to the target platoon
+         * 
+         * @param 
+         *   (double) front downtrack: the downtrack of the target platoon leader.
+         *   (double) rear downtrack: the downtrack of the last target platoon member.
+         *   (double) front crosstrack: the crosstrack of the target platoon leader. 
+         * @return
+         *   (bool): if the target leader behind
+         */
+
+       double currentDtd = current_downtrack_;
+       double currentCtd = current_crosstrack_;
+       // lateral error for two lanes 
+       double two_lane_cross_error = 2*config_.maxCrosstrackError + 3.5; // todo: 3.5m is lane width, consider read it from map
+       // todo: use maxgap (35m) as longitudinal threshold, consider change 
+       bool longitudinalCheck = ((currentDtd >= rearVehicleDtd - maxAllowedJoinGap_) || (currentDtd <= frontVehicleDtd + 10));
```

Comment:
Due to operator precedence, these two statements don't need any parentheses.  One could argue that the inner parens give you some comfort, but the outer set adds nothing but visual clutter.

Reply:
The value 10 seems quite arbitrary, and needs to be explained.  Better to get it from a config parameter.  Be sure the code can accommodate vehicles of various lengths (including heavy trucks).